src/kv/store.h

```diff
@@ -749,6 +749,14 @@ namespace kv
             return success;
           }
           auto h = get_history();
+          if (!h->verify(term_))
+          {
+            LOG_FAIL_FMT("Failed to deserialise");
+            LOG_DEBUG_FMT("Signature in transaction {} failed to verify", v);
+            throw std::logic_error(
+              "Failed to verify signature, view-changes not implemented");
+            return DeserialiseSuccess::FAILED;
```

Comment:
why is this returning just after throwing?

Reply:
Also what does signature verification have to do with view changes?

Reply:
It is throwing because i want the test to fail when this is happens.
In regards to view-changes, if the primary send a signature that does not verify we want to change the primary i.e. a view change which is not implemented yet